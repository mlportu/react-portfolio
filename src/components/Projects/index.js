import React, {useState} from 'react';
import Modal from '../Modal';

function Projects() {
    const [photos] = useState([
        {
            name:'Doggypedia',
            description: 'All the information you would want to know about any dog breed all in one place',
            link: 'https://kray306090.github.io/yorkie-terrier/'
        },
        {
            name: 'Health Log',
            description: 'Track and monitor your health to help identify healthy habbits to imporve your life',
            link: 'https://fathomless-thicket-41464.herokuapp.com/'
        },
        {
            name: 'Weather Dashboard',
            description: 'Know the weather for the next week in any city of your choice',
            link: 'https://mlportu.github.io/weather-dashboard/'
        },  
        {
            name: 'Workday Scheduler',
            description: 'Track your schedule and never miss an appointment',
            link: 'https://mlportu.github.io/workday-scheduler/'
        },
        {
            name: 'Tech Blog',
            description: 'Post all your favorite Tech news and updates to share with friends',
            link: 'https://frozen-fjord-11998.herokuapp.com/'
        },
        {
            name: 'Note Taker',
            description: 'Keep track of your thoughts and other notes throughout the day',
            link: 'https://vast-coast-30336.herokuapp.com/'
        },
    ])

    const [currentPhoto, setCurrentPhoto] = useState();
    const [isModalOpen, setIsModalOpen] = useState(false);

    const toggleModal = (image, i) => {
        setCurrentPhoto({...image, index:i});
        setIsModalOpen(true);
    }
    return(
        <div>
            {isModalOpen && <Modal currentPhoto={currentPhoto}></Modal>}
         
            <div className = "flex-row">
                {photos.map((image, i) => (
                    <img
                        src = {require(`../../assets/images/small/${i}.jpg`)}
                        alt={image.name}
                        className = "img-thumbnail mx-1"
                        onClick = {() => toggleModal(image, i)}
                        key = {image.name}
                        />
                ))}
            </div>
        </div>
    )
}

export default Projects;