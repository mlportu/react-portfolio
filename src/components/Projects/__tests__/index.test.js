import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Projects from '../'

afterEach(cleanup)

describe('PhotoList is rendering', () => {
  it('renders', () => {
    render(<Projects />);
  });

  it('renders', () => {
    const { asFragment } = render(<Projects />)
    expect(asFragment()).toMatchSnapshot()
  });
});