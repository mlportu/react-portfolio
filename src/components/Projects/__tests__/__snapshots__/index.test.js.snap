// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoList is rendering renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex-row"
    >
      <img
        alt="Doggypedia"
        class="img-thumbnail mx-1"
        src="0.jpg"
      />
      <img
        alt="Health Log"
        class="img-thumbnail mx-1"
        src="1.jpg"
      />
      <img
        alt="Weather Dashboard"
        class="img-thumbnail mx-1"
        src="2.jpg"
      />
      <img
        alt="Workday Scheduler"
        class="img-thumbnail mx-1"
        src="3.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
